--- neubot-0.4.16.8/neubot/main_win32.py.orig	Tue Oct 22 13:59:13 2013
+++ neubot-0.4.16.8/neubot/main_win32.py	Tue Oct 22 14:06:06 2013
@@ -32,11 +32,19 @@
 from neubot import utils_ctl
 from neubot import utils_version
 
-# Reference viewer and notifier so py2exe includes 'em
+# Reference modules to include them
 if sys.platform == 'win32' and not hasattr(sys, 'frozen'):
     from neubot import notifier
     from neubot import raw
     from neubot import viewer
+
+    from mod_dash import client_negotiate
+    from mod_dash import client_smpl
+    from mod_dash import main
+    from mod_dash import neubot_module
+    from mod_dash import server_glue
+    from mod_dash import server_negotiate
+    from mod_dash import server_smpl
 
 def subcommand_start(args):
     ''' Start subcommand '''
