From f2c68224bf331172082113312c9ea969b68cd223 Mon Sep 17 00:00:00 2001
From: Simone Basso <bassosimone@gmail.com>
Date: Thu, 13 Sep 2012 22:50:27 +0200
Subject: [PATCH 11/11] log: improve formatting of exceptions

---
 neubot/log.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/neubot/log.py b/neubot/log.py
index f622aa9..5f0b948 100644
--- a/neubot/log.py
+++ b/neubot/log.py
@@ -150,6 +150,10 @@ class StreamingLogger(object):
                                                       exc_info[1],
                                                       exc_info[2])
                 message = "%s\n%s\n" % (message, ''.join(exc_list))
+                for line in message.split('\n'):
+                    self._log(severity, line, None, None)
+                return
+
             message = message.rstrip()
 
             try:
@@ -269,6 +273,10 @@ class Logger(object):
                                                   exc_info[1],
                                                   exc_info[2])
             message = "%s\n%s\n" % (message, ''.join(exc_list))
+            for line in message.split('\n'):
+                self._log(severity, line, None, None)
+            return
+
         message = message.rstrip()
 
         # Write log into the database
-- 
Simone Basso

